django/db/models/functions/text.py

```diff
@@ -219,6 +219,60 @@ class RTrim(Transform):
     lookup_name = 'rtrim'
 
 
+class SHA1(Transform):
+    function = 'SHA1'
+    lookup_name = 'sha1'
+
+    def as_oracle(self, compiler, connection, **extra_content):
+        return super().as_sql(
+            compiler,
+            connection,
+            template=(
+                "LOWER(RAWTOHEX(STANDARD_HASH(UTL_I18N.STRING_TO_RAW("
+                "(%(expressions)s), 'AL32UTF8'), '%(function)s')))"
+            ),
+            **extra_content,
+        )
+
+    def as_postgresql(self, compiler, connection, **extra_content):
+        # Note that the digest() function requires the pgcrypto extension.
+        return super().as_sql(
+            compiler,
+            connection,
+            template="ENCODE(DIGEST((%(expressions)s), '%(function)s'), 'hex')",
+            function=self.function.lower(),
+            **extra_content,
+        )
+
+
+class SHA224(SHA1):  # FIXME: Not supported on Oracle?
```

Comment:
Looks like Oracle is the only one not to support this... Not sure how we'd handle that...

Reply:
As expected...
```
  test_basic (db_functions.text.test_sha224.SHA224Tests) ... ERROR (0.012s)
  test_transform (db_functions.text.test_sha224.SHA224Tests) ... ERROR (0.006s)
```

Reply:
It seems that there is no way to emulate `SHA224` on Oracle, even with [`DBMS_CRYPTO`](https://docs.oracle.com/en/database/oracle/oracle-database/18/arpls/DBMS_CRYPTO.html#GUID-CE3CF17D-E781-47CB-AEE7-19A9B2BCD3EC) :disappointed: .